### PR TITLE
use string instead of boolean

### PR DESCRIPTION
**Pull request by ghost-not-in-the-shell**
Explain your changes:
This PR uses string for compilation option for the network.

Explain how you tested your changes:
*

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

File: src/lib/signature_lib/schnorr.ml [hunk 1]
```diff
@@ -494,12 +450,16 @@ module Message = struct
 
   let network_id_testnet = Char.of_int_exn 0
 
+  let network_id_other = Char.of_int_exn 2
```
Comment (mrmr1993) [inline src/lib/signature_lib/schnorr.ml:453]: We want to achieve domain separation between each of these new domains too, but I don't think the mechanism is decided yet. Hard-coding 2 here is cryptographically dangerous.
Can you delete this, and put a `failwith "TODO"` where it's used below, to make sure this doesn't leak out into production?

File: src/lib/signature_lib/schnorr.ml [hunk 2]
```diff
@@ -494,12 +450,16 @@ module Message = struct
...
+
   let network_id =
     match Mina_signature_kind.t with
     | Mainnet ->
         network_id_mainnet
     | Testnet ->
         network_id_testnet
+    | Other_network _ ->
+        network_id_other
```
Comment (mrmr1993) [inline src/lib/signature_lib/schnorr.ml:462]: `failwith` here

File: src/lib/signature_lib/schnorr.ml [hunk 3]
```diff
@@ -524,7 +484,16 @@ module Message = struct
...
+    let derive ?(signature_kind = Mina_signature_kind.t) =
+      make_derive
+        ~network_id:
+          ( match signature_kind with
+          | Mainnet ->
+              network_id_mainnet
+          | Testnet ->
+              network_id_testnet
+          | Other_network _ ->
+              network_id_other )
```
Comment (mrmr1993) [inline src/lib/signature_lib/schnorr.ml:496]: And here

File: src/lib/signature_lib/schnorr.ml [hunk 4]
```diff
@@ -592,7 +564,16 @@ module Message = struct
...
```
Comment (mrmr1993) [inline src/lib/signature_lib/schnorr.ml:576]: Here also

**Review by mrmr1993 — CHANGES_REQUESTED** [review]

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

File: src/lib/hash_prefixes/hash_prefixes.ml [hunk 5]
```diff
@@ -48,6 +48,8 @@ let signature_testnet = create "CodaSignature"
 
 let signature_mainnet = create "MinaSignatureMainnet"
 
+let signature_other chain_name = create (chain_name ^ "Siganture")
```
Comment (bkase) [inline src/lib/hash_prefixes/hash_prefixes.ml:51]: nit: `Signature` not `Siganture`

**Review by mrmr1993 — APPROVED** [review]

File: src/lib/string_sign/string_sign.ml [hunk 6]
```diff
@@ -94,15 +94,43 @@ let%test_module "Sign_string tests" =
...
+    let%test "Sign with testnet, fail to verify with mainnet or other networks"
+        =
       let open Mina_signature_kind in
       let s = "Some pills make you larger" in
       let signature = sign ~signature_kind:Testnet keypair.private_key s in
-      not (verify ~signature_kind:Mainnet signature keypair.public_key s)
+      (not (verify ~signature_kind:Mainnet signature keypair.public_key s))
+      && not
+           (verify ~signature_kind:(Other_network "Foo") signature
+              keypair.public_key s )
```
Comment (deepthiskumar) [inline src/lib/string_sign/string_sign.ml:113]: For this and all the below tests, could you also add a check that they can be verified with the intended signature kind?

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Pull request merged**